lib/image-diff.js

```diff
@@ -83,9 +83,18 @@ class ImageDiff {
         const {
           x, y, w, h,
         } = getBlockOutArea(dstImage, area, dpi);
-
-        fillPngRect(aImage, x, y, w, h);
-        fillPngRect(bImage, x, y, w, h);
+        if (w && h) {
```

Comment:
I would opt for a single action to cover both use cases, otherwise we would run `fillPngRect` twice, isn't?

Reply:
I'm not sure what you are referring to.
this `if (w && h) {` ?

Reply:
I'm just adding some safety check here to avoid filling blocked elements that are outside the bounds, which is causing `fillPngRect()` to crash during the diff.
If those elements are in different place in each images, or the images have different sizes, then it will fail as designed (image diff) and it will be visible in the report.

Reply:
actually I just realized what you meant, because there is a single blockOut.json record.
shouldn't we generate a blockOut.json per target (target passed to --take-snapshot)?

Reply:
drafted the blockOut split per snapshot target here #52 
that also address #49 - so would replace this PR